src/GNDStk/BodyText.hpp

```diff
@@ -2,10 +2,10 @@
 // For printing.
 // When writing a Component with its generic write() function (or its stream
 // output, which uses write()), and the Component is based on a BodyText<true>,
-// values will be printed in GNDStk::columns columns. "columns" is aliased to
+// values will be printed with GNDStk::columns across. "columns" is aliased to
 // "across" for convenience, because, at the time of this writing, GNDStk has
 // a Meta<> object, named "columns", which would also be in scope if the core
-// namespace is used. So, a user might prefer to use "across".
+// namespace is used. So, a user might prefer to use the name "across".
 inline std::size_t columns = 4;
 inline std::size_t &across = columns;
```

Comment:
My I/O system is a good one, and very carefully thought out. What I perhaps should do, at some point, is to consolidate all printing-related parameters into their own sub-namespace within `GNDStk::`, and put in one place. At the moment, some of the printing-related parameters were added recently, because they related to new features, and were placed where it was convenient to do so. But I can look at this at some point, locate anything in this regard, and consolidate it all in one place. They, I think it'll be clearer why I'm doing things this way.

GNDStk's overall I/O system, and in particular its printing capabilities, applies to at least the following: print XML, print JSON, print a Node in an internal format that's useful for debugging, and print objects that are derived from Component. Things like `GNDStk::indent` are sort of like environment variables in linux: set it to what you like, and then it applies to whatever uses it, so you no longer need to clutter individual print commands with stuff like that. Set it once, instead of many times.

C++ does not, at present, allow additional arguments to things like `operator<<`. Even if it did, using it in such a way would probably require that someone uses the awkward functional form of the call; something like `std::cout.operator<<(value,then,some,additional,arguments)`. Instead of something nice and short like `std::cout << value`, with "environment variables" set up beforehand, the way you like them, and automatically used in any `<<` invocation that happens to respect the environment variable. Coloring (for errors, warnings, and the like; and also for nice color-coded output of objects derived from Component), and even whether or not coloring is switched on, is also something for which these convenient global settings can be made.

We all learn early in our computer science classes that "global variables are bad." :-) For many things, they are. However, also like many things, when used judiciously they can be a good thing. Most things are - appropriately - function parameters. However, over the years I've developed a nice I/O system that allow certain niceties (basically, pretty-printing issues like indentation and colorization) to be set up with namespace-scope variables (namespaces are made for polluting!!), to set user preferences that they probably want *globally*. Then, individual prints and short and sweet. With several parameters available, default argument lists (if they were even possible with things like `operator<<`) might require that several "defaults" be given explicitly in order to get to one default, later in the argument list, that someone wants to change. (Because as you know, C++ isn't flexible with argument ordering, as Python is.)

Things that should still be parameters, are parameters. An example is indentation level. (Not to be confused with GNDStk::indent, which is the number of spaces you like, PER level. I like 3, you prefer 2.) For example, `Component::write()` (which automatically imbues its derived class with a sophisticated, customizable, and colorizable print capability) has a level that defaults to 0. It also provided an `operator<<`, which doesn't (and can't, per the C++ standard) allow level to be set. So, `operator<<` calls `write()` with `level=0`. Once inside the `write()` ecosystem, the code calls other `write()` functions, with each dealing with `level` in a nice way. The result will be, for example, that users will have a SUPER nice and flexible printing capability for absolutely *every* GNDS Standard Interface class - all those things that derive from Component. That is, Component provides very generic pretty-printing capabilities, and the "environment variables" that control stylistic aspects of printing are available, separately, for anyone who wishes to use them. Users get a whole lot, for relatively little effort on our part.

This general system is something I've developed over the years, and it's very easy-to-use, effective, and substantially better than other alternatives I've seen.